### PR TITLE
stone-kingdoms: init at 0.5.0

### PR DESCRIPTION
**Pull request by hulr**
###### Description of changes

https://gitlab.com/stone-kingdoms/stone-kingdoms

A real-time strategy game made with LÖVE based on the original Stronghold by Firefly studios.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
Result of `nixpkgs-review pr 240975` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stone-kingdoms</li>
  </ul>
</details>

**Comment by teto** [comment]
tried the game, looks great. Game resources are apache too ? These games tend to use the original game assets, can you confirm it's not the case ?

**Comment by hulr** [comment]
On Flathub it is also listed with Apache License 2.0: https://flathub.org/apps/io.gitlab.stone_kingdoms.StoneKingdoms
Stone Kingdom uses image assets which are property of Firefly Studios according to the README: https://gitlab.com/stone-kingdoms/stone-kingdoms/-/blob/0.5.0/README.md

**Comment by teto** [comment]
are they allowed to use the assets mentioned at https://gitlab.com/stone-kingdoms/stone-kingdoms/-/blob/0.5.0/assets/tiles/LICENSE ? https://fireflyworlds.com/ still exists

**Review by AndersonTorres — DISMISSED** [review]

**Comment by hulr** [comment]
@teto I don't know if they are allowed to use these assets. Could you maybe contact the developers over at GitLab if you are concerned about the license?
I think this would be a general issue and not just related to the Nix package.

**Comment by teto** [comment]
I dont play the game so it should be easier for you to contact them.
The license is an issue if you wanna merge this in nixpkgs, having a PR is already great for people to try out the game

**Comment by AndersonTorres** [comment]
Can you split the assets to another expression file?

**Comment by hulr** [comment]
@AndersonTorres What is the benefit of using two expression files? Do you have an example?
Wouldn't it be enough to specify two licenses like this:
`license = with licenses; [ asl20 unfree ];`

**Comment by AndersonTorres** [comment]
First, unfree things aren't built by Hydra, therefore the package would not be cached.

Second, my idea is that possibly the assets can be substituted for other assets, like Doom games, where you can download other assets and run them on the same engine.

Therefore, it is not interesting to force the "vendoring" of the unfree assets.

**Comment by hulr** [comment]
Thanks for your explanations.
I had a look at the source code and I don't think it makes sense to split up the assets to another expression file. The game itself and assets are tightly coupled.
To my understanding, quads are created in `object_quads.lua` based on images in `stronghold_assets_packed_v8-hd.dds` and those quads are then referenced throughout the game. 
See the following commit: https://gitlab.com/stone-kingdoms/stone-kingdoms/-/commit/8f483dc851788532e462c3b1cc34894153570054

**Comment by AndersonTorres** [comment]
Well, then there is nothing to be done, after all.

File: pkgs/games/stone-kingdoms/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    makeWrapper ${love}/bin/love $out/bin/stone-kingdoms \
+      --add-flags $out/share/games/lovegames/stone-kingdoms.love
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A real-time strategy game made with LÖVE based on the original Stronghold by Firefly studios";
+    homepage = "https://gitlab.com/stone-kingdoms/stone-kingdoms";
+    platforms = platforms.linux;
+    license = licenses.asl20;
```
Comment (AndersonTorres) [inline pkgs/games/stone-kingdoms/default.nix:57]: Given what was said, 
```suggestion
    license = with licenses; [ 
      asl20 # engine
      unfree # game assets 
    ];
```

**Comment by PedroHLC** [comment]
> Well, then there is nothing to be done, after all.

The said project is fully interpreted. We won't save any building time anyway.

But maybe we can use `stdenvNoCC` to save a few bucks.

**Review by PedroHLC — APPROVED** [review]

**Pull request merged**